interpreter/cling/lib/Interpreter/ValuePrinter.cpp

```diff
@@ -457,26 +221,31 @@ static std::string getCastValueString(const Value &V) {
   return strm.str();
 }
 
-
 static std::string printEnumValue(const Value &V){
   std::stringstream enumString;
   clang::ASTContext& C = V.getASTContext();
   clang::QualType Ty = V.getType().getDesugaredType(C);
-  clang::EnumDecl* ED = Ty.getNonReferenceType()->getAs<clang::EnumType>()->getDecl();
-  uint64_t value = *(const uint64_t*)&V;
-  bool IsFirst = true;
-  llvm::APSInt ValAsAPSInt = C.MakeIntValue(value, Ty);
-  for (clang::EnumDecl::enumerator_iterator I = ED->enumerator_begin(),
-           E = ED->enumerator_end(); I != E; ++I) {
-    if (I->getInitVal() == ValAsAPSInt) {
-      if (!IsFirst) {
-        enumString << " ? ";
+  const clang::EnumType *EnumTy = Ty.getNonReferenceType()->getAs<clang::EnumType>();
+  if(EnumTy != 0) {
```

Comment:
This was a fix to the bug that Coverity found - https://coverity.cern.ch/reports.htm#v13142/p10001
If I understand the code correctly, <code>getAs</code> should return <code>nullptr</code> in case of a failure (http://clang.llvm.org/doxygen/Type_8h_source.html#l05545)?